pkg/descheduler/controllers/migration/controller.go

```diff
@@ -787,6 +791,22 @@ func (r *Reconciler) waitForPodBindReservation(ctx context.Context, job *sev1alp
 	return true, reconcile.Result{}, nil
 }
 
+func (r *Reconciler) waitForPodReady(ctx context.Context, podNamespacedName types.NamespacedName) (reconcile.Result, error) {
+	klog.V(4).Infof("checks if the boundPod %q is ready or not", podNamespacedName)
+	podObj := &corev1.Pod{}
+	if err := r.Client.Get(ctx, podNamespacedName, podObj); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{RequeueAfter: defaultRequeueAfter}, fmt.Errorf("failed to get pod %q", podNamespacedName)
+	}
+	if isReady := k8spodutil.IsPodReady(podObj); !isReady {
+		return reconcile.Result{RequeueAfter: defaultRequeueAfter}, fmt.Errorf("boundPod: %q is not ready", podNamespacedName)
```

Comment:
Maybe we can add event for the MigrationJob indicates that it is waiting for Pod ready, otherwise the user may think that MigrationJob hangs because the Pod has not been ready for a long time. In addition, a type can also be added to the Job Status Condition to indicate whether the Pod is ready

Reply:
done